src/joystick/joystick_manager.cpp

```diff
@@ -0,0 +1,301 @@
+
+    // If this ever starts throwing weird compile time errors about
+    // exe_cmd_init being deleted, just go to posix/basic_cmd.hpp
+    // and mark the exe_cmd_init(const ...) function as default
+    // See: https://github.com/klemens-morgenstern/boost-process/issues/21
+
+    process = ::boost::process::child(
+            ::boost::process::search_path("roslaunch"),
+            "roboteam_input",
+            "joy_node.launch",
+            "jsTarget:=" + input,
+            "deadzone:=0.20",
+            "nodeName:=JoyStick_" + this->input,
+            ::boost::process::std_out > ::boost::process::null
+    );
+
+    // Make new subscriber
+    ros::NodeHandle n;
+    subscriber = n.subscribe(input, 1, &JoyEntry::receiveJoyMsg, this);
+}
+
+void JoyEntry::receiveJoyMsg(const sensor_msgs::JoyConstPtr &msg) {
+    //std::cout<<"Do stuff"<<std::endl;
+    //this->msg_prev = this->msg;     // Store the current message in previous message
+    this->msg = *msg;               // Only store the newest messages
+}
+
+// ==== Set robot ID and initialize speed and geneva state ==== //
+void JoyEntry::setRobotID(int id){
+    ROS_INFO_STREAM(input << " connected to robot " << id);
+    robotID = id;
+    // Reset robot velocity
+    speedState.x = 0;
+    speedState.y = 0;
+    // Reset geneva drive
+    genevaState = 3;
+    // Reset the orientation
+    orientation = 0.0;
+}
+
+// ==== Switch to next joystick profile ==== //
+void JoyEntry::nextJoystickProfile(){
+    profileCounter = (profileCounter + 1) % NUM_JOYSTICK_PROFILES;
+    profile = joystick_profiles[profileCounter];
+    ROS_INFO_STREAM(input << " using profile " << profileCounter << " " << joystick_profiles[profileCounter].DESCRIPTION);
+}
+
+// ==== Switch control mode ==== //
+void JoyEntry::switchControlMode(){
+    if(useRelativeControl) {
+        useRelativeControl = false;
+        ROS_INFO_STREAM(input << " using control mode absolute (Fifa)");
+    } else {
+        useRelativeControl = true;
+        ROS_INFO_STREAM(input << " using control mode relative (Call of Duty)");
+    }
+}
+
+// ==== Check whether joystick is connected ==== //
+void JoyEntry::setControllerConnected(bool isConnected){
+    if(isConnected && !controllerConnected)
+        ROS_INFO_STREAM(input << ": Controller reconnected");
+
+    if(!isConnected && controllerConnected)
+        ROS_WARN_STREAM(input << ": Controller disconnected");
+
+    controllerConnected = isConnected;
+
+    // Clear message if controller is disconnected, to make the robot stop
+    if(!controllerConnected)
+        msg = boost::none;
+        msg_prev = boost::none;
+
+}
+
+template<typename T> T getVal(const std::vector<T> &values, int index) {
+    if(index < values.size()) {
+        return values[index];
+    }
+    return T(0);
+}
+
+// ==== Handle buttons such as control mode and ID, and geneva state ==== //
+void handleButtons(JoyEntry &joy, sensor_msgs::Joy const &msg, sensor_msgs::Joy const &msg_prev){
```

Comment:
msg_prev does not obey the naming convention. it should be msgPrev, or even better: previousMsg

Reply:
Cool, will do!